core/app/models/spree/store_credit.rb

```diff
@@ -0,0 +1,253 @@
+        false
+      end
+    end
+
+    def actions
+      [CAPTURE_ACTION, VOID_ACTION, CREDIT_ACTION]
+    end
+
+    def can_capture?(payment)
+      payment.pending? || payment.checkout?
+    end
+
+    def can_void?(payment)
+      payment.pending?
+    end
+
+    def can_credit?(payment)
+      payment.completed? && payment.credit_allowed > 0
+    end
+
+    def generate_authorization_code
+      "#{self.id}-SC-#{Time.now.utc.strftime("%Y%m%d%H%M%S%6N")}"
+    end
+
+    class << self
+      def default_created_by
+        Spree.user_class.find_by(email: DEFAULT_CREATED_BY_EMAIL)
+      end
+    end
+
+    private
+
+    def create_credit_record(amount, action_attributes={})
+      # Setting credit_to_new_allocation to true will create a new allocation anytime #credit is called
+      # If it is not set, it will update the store credit's amount in place
+      credit = nil
+      if Spree::Config.credit_to_new_allocation
+        credit = Spree::StoreCredit.new(create_credit_record_params(amount))
+      else
+        self.amount_used = amount_used - amount
+        credit = self
+      end
+
+      credit.assign_attributes(action_attributes)
+      credit.save!
+    end
+
+    def create_credit_record_params(amount)
+      {
+        amount: amount,
+        user_id: self.user_id,
+        category_id: self.category_id,
+        created_by_id: self.created_by_id,
+        currency: self.currency,
+        type_id: self.type_id,
+        memo: credit_allocation_memo,
+      }
+    end
+
+    def credit_allocation_memo
+      "This is a credit from store credit ID #{self.id}"
+    end
+
+    def store_event
+      return unless amount_changed? || amount_used_changed? || amount_authorized_changed? || action == ELIGIBLE_ACTION
+
+      event = if action
+        store_credit_events.build(action: action)
+      else
+        store_credit_events.where(action: ALLOCATION_ACTION).first_or_initialize
+      end
+
+      event.update_attributes!(
+        {
+          amount: action_amount || amount,
+          authorization_code: action_authorization_code || event.authorization_code || generate_authorization_code,
+          user_total_amount: user.total_available_store_credit,
+          originator: action_originator,
+        }
+      )
+    end
+
+    def amount_used_less_than_or_equal_to_amount
+      return true if amount_used.nil?
+
+      if amount_used > amount
+        errors.add(:amount_used, Spree.t('admin.store_credits.errors.amount_used_cannot_be_greater'))
+      end
+    end
+
+    def amount_authorized_less_than_or_equal_to_amount
+      if (amount_used + amount_authorized) > amount
+        errors.add(:amount_authorized, Spree.t('admin.store_credits.errors.amount_authorized_exceeds_total_credit'))
+      end
+    end
+
+    def validate_no_amount_used
+      if amount_used > 0
+        errors.add(:amount_used, 'is greater than zero. Can not delete store credit')
+      end
+    end
```

Comment:
This method should return false, or else the destroy will go ahead. You could do it like this:

```ruby
def validate_no_amount_used
  if amount_used > 0
    errors.add(...')
  end
  errors.blank?
end
```